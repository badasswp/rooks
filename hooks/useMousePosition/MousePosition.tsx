import type { JSX } from 'react';
import useMousePosition from './';

/**
 * MousePosition component.
 *
 * This component is used to demonstrate the
 * use of the useMousePosition hook.
 *
 * @returns {JSX.Element} The MousePosition component.
 */
const MousePosition = (): JSX.Element => {
  const {x, y} = useMousePosition();

  return (
    <div>
      <p>Try to resize your Window</p>
      <p data-testid="pos-x">Your current mouse position on the x-axis is {x}</p>
      <p data-testid="pos-y">Your current mouse position on the y-axis is {y}</p>
    </div>
  );
}

export default MousePosition;
