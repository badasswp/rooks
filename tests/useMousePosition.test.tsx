import '@testing-library/jest-dom';
import { getByTestId, render, waitFor } from '@testing-library/react';

import MousePosition from '../hooks/useMousePosition/MousePosition';

describe('<UseMousePosition />', () => {
  it('should render the component with no issues', () => {
    const { container } = render(<MousePosition />);

    expect(container).toMatchSnapshot();
  });

  it('should display the x and y positions using the useMousePosition hook', async () => {
    const { container } = render(<MousePosition />);

    const mouseMoveEvent = new MouseEvent('mousemove', {
      bubbles: true,
      clientX: 300,
      clientY: 200,
    });

    window.dispatchEvent(mouseMoveEvent);

    await waitFor(() => {
      expect(getByTestId(container, 'pos-x')).toHaveTextContent(
        'Your current mouse position on the x-axis is 300'
      );
      expect(getByTestId(container, 'pos-y')).toHaveTextContent(
        'Your current mouse position on the y-axis is 200'
      );
    });
  });
});
