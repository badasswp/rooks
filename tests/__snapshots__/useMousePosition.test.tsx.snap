// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseMousePosition /> should render the component with no issues 1`] = `
<div>
  <div>
    <p>
      Try to resize your Window
    </p>
    <p
      data-testid="pos-x"
    >
      Your current mouse position on the x-axis is 
      0
    </p>
    <p
      data-testid="pos-y"
    >
      Your current mouse position on the y-axis is 
      0
    </p>
  </div>
</div>
`;
